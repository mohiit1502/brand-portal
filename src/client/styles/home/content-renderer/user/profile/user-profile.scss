@import "../../../../override/variables";

.user-profile-content {
  .content-header-row {
    font-weight: bold;
  }
  .password-reset-col{
    button{
      border-radius: 42px;
      font-weight: bold;
      font-size: 15px;

      &.reset-password {
        border: $default-input-border;
        color: $market-place-blue;
      }

      &.manage-profile {
        color: white;
      }

      &:hover {
        color: white;
        background: $market-place-blue;
      }
    }
  }

  .edit-button-panel {
    .btn-primary{
      border-radius: 42px;
      background: #006799;
      font-weight: bold;
      font-size: 15px;
      border: none;
    }
    .btn-link{
      font-weight: bold;
      font-size: 18px;
      text-decoration: underline;
    }
  }
  .password-reset-row {
    .badge {
      //border: 1px solid $blue;
    }
  }
}
