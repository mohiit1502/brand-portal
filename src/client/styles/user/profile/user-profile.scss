@import "../../override/variables";

.user-profile-content {
  .password-reset-row {
    .badge {
      border: 1px solid $blue;
    }
  }
}
