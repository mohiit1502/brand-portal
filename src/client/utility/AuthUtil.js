import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
