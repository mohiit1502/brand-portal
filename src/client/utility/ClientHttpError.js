export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
