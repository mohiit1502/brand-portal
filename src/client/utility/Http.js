/* eslint-disable complexity */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-params */
/* eslint-disable max-statements */
import fetch from "node-fetch";
import queryString from "query-string";
import ClientHttpError from "./ClientHttpError";
import { NOTIFICATION_TYPE } from "../actions/notification/notification-actions";
import CONSTANTS from "../constants/constants";
export default class Http {

  static async get(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});

    const options = {
      method: "GET"
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;

    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message, err.code);
  }

  static async post(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async postAsFormData(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: data,
      headers: { }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async put(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "PUT",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };

    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);
  }

  static async delete(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "DELETE",
      noInject: true
    };
    return fetch(urlString, options);
  }

  displayToast(status, toastcallback) {
    if (new RegExp(CONSTANTS.CODES.ERRORCODES.SERVERERROR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Unable to reach our services currently, please try again in sometime.`);
    } else if (new RegExp(CONSTANTS.CODES.ERRORCODES.FOURNOTFOUR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Requested Resource could not found.`);
    }
  }
}
