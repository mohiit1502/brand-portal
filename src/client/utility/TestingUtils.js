import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
