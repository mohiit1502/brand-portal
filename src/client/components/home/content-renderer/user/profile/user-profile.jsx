import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Http from "../../../../../utility/Http";
import {updateUserProfile} from "../../../../../actions/user/user-actions";
import {dispatchDiscardChanges, TOGGLE_ACTIONS, toggleModal} from "../../../../../actions/modal-actions"
import {showNotification} from "../../../../../actions/notification/notification-actions"
import InputFormatter from "../../../../../utility/phoneOps";
import Helper from "../../../../../utility/helper";
import ContentRenderer from "../../../../../utility/ContentRenderer";
import Validator from "../../../../../utility/validationUtil";
// import FORMFIELDCONFIG from "./../../../../../config/formsConfig/form-field-meta";
import "../../../../../styles/home/content-renderer/user/profile/user-profile.scss";
import CONSTANTS from "../../../../../constants/constants";
import mixpanel from "../../../../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../../../../constants/mixpanelConstants";
import Cookies from "electrode-cookies";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    const functions = ["bubbleValue", "displayChangePassword", "displayManageProfileNotification", "isDirty", "onChange", "setFormData", "disableInput", "saveUser"];
    functions.forEach(name => this[name] = this[name].bind(this));
    this.validateState = Validator.validateState.bind(this);
    this.getFieldRenders = ContentRenderer.getFieldRenders.bind(this);
    this.loader = Helper.loader.bind(this);
    this.onInvalid = Validator.onInvalid.bind(this);
    this.invalid = {firstName: false, lastName: false, companyName: false, phone: false};
    const userProfileConfiguration = this.props.userProfileContent ? this.props.userProfileContent : {}

    this.state = {
      section: {...userProfileConfiguration.sectionConfig},
      form: {
        ...userProfileConfiguration.formConfig,
        inputData: {...userProfileConfiguration.fields},
        // underwritingChecked: false,
      },
      isSeller: Cookies.get("client_type") === "seller"
    };

    const formatter = new InputFormatter();
    const handlers = formatter.on(`#${this.state.section.id}-${this.state.form.inputData.phone.inputId}-custom-input`);
    this.prebounceChangeHandler = handlers.inputHandler;

    Object.keys(this.state.form.inputData).forEach(itemKey => {
      const item = this.state.form.inputData[itemKey];
      this.state.form.inputData[itemKey].value = Helper.search(item.initValuePath, this.props.userProfile);
      if  (itemKey === "phone") {
        const value = this.state.form.inputData[itemKey].value;
        this.state.form.inputData[itemKey].value = (value === "0000000000") || (value === "(000) 000-0000") ? "" : value;
      }
    });
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setFormData(this.props.userProfile);
    }
  }

  isDirty () {
    const {firstName: originalFName, lastName: originalLName, email: originalEmail, phoneNumber: originalPhone, companyName: originalCompany} = this.props.userProfile;
    const {firstName: {value: currentFName}, lastName: {value: currentLName}, emailId: {value: currentEmail}, phone: {value: currentPhone}} = this.state.form.inputData;
    const currentCompany = this.state.form.inputData.company ? this.state.form.inputData.company.value : "";
    const originalPhoneModified = (originalPhone === "0000000000" || originalPhone === "(000) 000-0000") ? "" : originalPhone;
    return originalFName !== currentFName || originalLName !== currentLName || originalEmail !== currentEmail || originalPhoneModified !== currentPhone || (originalCompany && currentCompany && originalCompany !== currentCompany);
      // || ((!originalCompany && currentCompany) || (originalCompany && !currentCompany) || (originalCompany !== currentCompany));
  }

  setFormData(obj) {
    if (this.props.shouldDiscard) {
      const form = {...this.state.form};
      form.isDisabled = true;
      form.inputData.firstName.value = obj.firstName;
      form.inputData.lastName.value = obj.lastName;
      form.inputData.companyName.value = obj.type === "ThirdParty" ? obj.companyName : "";
      form.inputData.emailId.value = obj.email;
      form.inputData.phone.value = (obj.phoneNumber === "0000000000") || (obj.phoneNumber === "(000) 000-0000") ? "" : obj.phoneNumber;

      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = true;
          item.error = "";
        }
      });

      this.setState({form});
      this.props.dispatchDiscardChanges(false);
    }
  }

  bubbleValue (evt, key, error) {
    const targetVal = evt.target.value;
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      state.form.inputData[key].error = error;
      return state;
    });
  }

  onChange (evt, key) {
    if (evt && evt.target) {
      if (evt.target.pattern && evt.target.checkValidity()) {
        this.invalid[key] = false;
      }
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        const inputData = state.form.inputData;
        state.form.inputData[key].value = targetVal;
        inputData[key].error = !this.invalid[key] ? "" : inputData[key].error;
        this.invalid[key] = false;
        return {
          ...state
        };
      });
    }
  }

  displayManageProfileNotification () {
    let template = this.props.modalsMeta && this.props.modalsMeta.PASSWORD_RESET_SELLER;
    template = {templateName: "StatusModalTemplate", image: "", ...template};
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...template});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_SELLER_MANAGE_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  displayChangePassword() {
    const meta = { templateName: "ResetPasswordTemplate" };
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_CHANGE_PASSWORD, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  disableInput (disable) {
    disable = !!disable;
    if (disable && this.isDirty()) {
      const meta = { templateName: "Alert" };
      this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
      mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.CANCEL_EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    } else {
      const form = {...this.state.form};
      form.isDisabled = disable;
      form.inputData.phone.disabled = disable;
      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = disable;
          item.error = "";
        }
      });
      this.setState({form});
      if (!disable) mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    }
  }

  // toggleUnderwritingCheck () {
  //   const form = {...this.state.form};
  //   form.underwritingChecked = !form.underwritingChecked;
  //   this.setState({form});
  // }

  async saveUser (evt) {
    evt.preventDefault();

    if (!this.validateState()) {
      this.loader("form", true);
      const loginId = this.state.form.inputData.emailId.value;
      const firstName = this.state.form.inputData.firstName.value;
      const lastName = this.state.form.inputData.lastName.value;
      const phoneNumber = this.state.form.inputData.phone.value ? this.state.form.inputData.phone.value : "0000000000"; //[note:to handle VIP phone number validation]
      const payload = {
        user: {
          loginId,
          firstName,
          lastName,
          phoneNumber,
          properties: {
            companyName: this.state.form.inputData.companyName.value
          }
        }
      };

      const url = this.state.form.apiPath;
      if (this.isDirty()) {
        const mixpanelPayload = {
          API: url,
          WORK_FLOW: "EDIT_USER_PROFILE"
        };
        return Http.put(`${url}/${payload.user.loginId}`, payload, null, null, this.props.showNotification, this.state.form.profileSaveMessage)
          .then(async res => {
            this.loader("form", false);
            this.props.updateUserProfile(res.body);
            this.disableInput(true);
            mixpanelPayload.API_SUCCESS = true;
          })
          .catch(err => {
            this.loader("form", false);
            mixpanelPayload.API_SUCCESS = false;
            mixpanelPayload.ERROR = err.message ? err.message : err;
          })
          .finally(() => {
            mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.SAVE_PROFILE, mixpanelPayload);
          });
      } else {
        this.loader("form", false);
        this.disableInput(true);
      }
    }
    return null;
  }

  render () {
    return (
      <div className={`user-profile-content h-100${this.state.form.loader ? " loader" : ""}`}>
        <header className={this.state.section.headerClasses}>{this.state.section.sectionTitle}</header>
        <form className={this.state.section.formClasses} autoComplete="off" onSubmit={this.saveUser} style={{marginRight: "50%"}}>
          {this.getFieldRenders()}
        </form>
      </div>
    );
  }
}

UserProfile.propTypes = {
  dispatchDiscardChanges: PropTypes.func,
  shouldDiscard: PropTypes.bool,
  showNotification: PropTypes.func,
  toggleModal: PropTypes.func,
  userProfile: PropTypes.object,
  updateUserProfile: PropTypes.func
};

const mapStateToProps = state => {
  return {
    modalsMeta: state.content.metadata ? state.content.metadata.MODALSCONFIG : {},
    userProfileContent: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.USERPROFILE,
    shouldDiscard: state.modal.shouldDiscard,
    userProfile: state.user.profile
  };
};

const mapDispatchToProps = {
  dispatchDiscardChanges,
  showNotification,
  toggleModal,
  updateUserProfile
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserProfile);
