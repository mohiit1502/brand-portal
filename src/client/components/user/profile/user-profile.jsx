import React from "react";
import { connect } from "react-redux";
import "../../../styles/user/profile/user-profile.scss";
import PropTypes from "prop-types";
import CustomInput from "../../custom-input/custom-input";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.setFormData = this.setFormData.bind(this);
    this.disableInput = this.disableInput.bind(this);
    this.toggleUnderwritingCheck = this.toggleUnderwritingCheck.bind(this);

    this.state = {
      form: {
        isDisabled: true,
        underwritingChecked: false,
        id: "user-profile-form",
        inputData: {
          firstName: {
            label: "First Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          lastName: {
            label: "Last Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          companyName: {
            label: "Company Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          emailId: {
            label: "Email ID",
            required: true,
            value: "",
            type: "email",
            pattern: null,
            disabled: true
          },
          phone: {
            label: "Mobile Number",
            required: false,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          }
        }
      }
    };
  }

  componentDidMount() {
    const obj = {
      firstName: "Shubhansh",
      lastName: "Sahai",
      companyName: "Walmart Labs",
      emailId: "shubhansh.sahai@walmartlabs.com",
      phone: "9686648597"
    };

    this.setFormData(obj);

  }

  setFormData(obj) {
    const form = {...this.state.form};
    form.inputData.firstName.value = obj.firstName;
    form.inputData.firstName.disabled = true;
    form.inputData.lastName.value = obj.lastName;
    form.inputData.lastName.disabled = true;
    form.inputData.companyName.value = obj.companyName;
    form.inputData.companyName.disabled = true;
    form.inputData.emailId.value = obj.emailId;
    form.inputData.emailId.disabled = true;
    form.inputData.phone.value = obj.phone;
    form.inputData.phone.disabled = true;

    this.setState({
      form
    });
  }

  onInputChange (evt, key) {
    if (evt && evt.target) {
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        state.form.inputData[key].value = targetVal;
        return {
          ...state
        };
      });
    }
  }

  disableInput (disable) {
    disable = !!disable;

    const form = {...this.state.form};

    form.isDisabled = disable;

    form.inputData.firstName.disabled = disable;
    form.inputData.lastName.disabled = disable;
    form.inputData.companyName.disabled = disable;
    form.inputData.emailId.disabled = disable;
    form.inputData.phone.disabled = disable;

    this.setState({
      form
    });

  }

  toggleUnderwritingCheck () {
    const form = {...this.state.form};
    form.underwritingChecked = !form.underwritingChecked;
    this.setState({form});
  }

  render () {

    return (
      <div className="row user-profile-content h-100">
        <div className="col h-100">
          <div className="row h-10">
            <div className="col">
              <h3>User Profile</h3>
            </div>
          </div>
          <div className="row mt-4 h-90">
            <div className="col">
              <form className="h-100" autoComplete="off">
                <div className="row h-60">
                  <div className="col h-100">
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"firstName"}
                          inputId={"firstName"}
                          formId={this.state.form.id} label={this.state.form.inputData.firstName.label}
                          required={this.state.form.inputData.firstName.required} value={this.state.form.inputData.firstName.value}
                          type={this.state.form.inputData.firstName.type} pattern={this.state.form.inputData.firstName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.firstName.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"lastName"}
                          inputId={"lastName"}
                          formId={this.state.form.id} label={this.state.form.inputData.lastName.label}
                          required={this.state.form.inputData.lastName.required} value={this.state.form.inputData.lastName.value}
                          type={this.state.form.inputData.lastName.type} pattern={this.state.form.inputData.lastName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.lastName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"companyName"}
                          inputId={"companyName"}
                          formId={this.state.form.id} label={this.state.form.inputData.companyName.label}
                          required={this.state.form.inputData.companyName.required} value={this.state.form.inputData.companyName.value}
                          type={this.state.form.inputData.companyName.type} pattern={this.state.form.inputData.companyName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.companyName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"emailId"}
                          inputId={"emailId"}
                          formId={this.state.form.id} label={this.state.form.inputData.emailId.label}
                          required={this.state.form.inputData.emailId.required} value={this.state.form.inputData.emailId.value}
                          type={this.state.form.inputData.emailId.type} pattern={this.state.form.inputData.emailId.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.emailId.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"phone"}
                          inputId={"phone"}
                          formId={this.state.form.id} label={this.state.form.inputData.phone.label}
                          required={this.state.form.inputData.phone.required} value={this.state.form.inputData.phone.value}
                          type={this.state.form.inputData.phone.type} pattern={this.state.form.inputData.phone.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.phone.disabled} />
                      </div>
                    </div>
                    <div className="row mb-5 password-reset-row">
                      <div className="col-3">
                        <div className="btn btn-primary btn-sm no-border-radius px-3">
                          Password Reset
                        </div>
                        <span className="badge badge-pill bg-white text-blue ml-2 cursor-pointer"> ? </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div className="row h-40">
                  <div className="col h-100">
                    <div className="row pt-5">
                      <div className="col-xl-6 col-10">
                        <label>
                          <input type="checkbox" className="mr-2" checked={this.state.form.underwritingChecked} onChange={this.toggleUnderwritingCheck}/>
                          I have a good faith belief that the use of the material in the manner complained of is not authorized by the copyright owner, its agent, or the law.
                        </label>
                      </div>
                    </div>
                    <div className="row pt-5">
                      <div className="col-xl-3 col-5" />
                      <div className="col-xl-3 col-5 text-right">
                        {
                          this.state.form.isDisabled ?
                            (
                              <div className="btn btn-primary btn-sm no-border-radius px-4" onClick={() => this.disableInput(false)}>
                                Edit
                              </div>
                            ) :
                            (
                              <span>
                                <div className="btn btn-link font-size-14 px-4 mr-3" onClick={() => this.disableInput(true)}>Cancel</div>
                                <button type="submit" className="btn btn-primary btn-sm no-border-radius px-4 font-size-14" disabled={!this.state.form.underwritingChecked}>
                                  Save
                                </button>
                              </span>
                            )
                        }

                      </div>
                    </div>
                  </div>
                </div>

              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => state;

export default connect(
  mapStateToProps,
  dispatch => ({dispatch})
)(UserProfile);
