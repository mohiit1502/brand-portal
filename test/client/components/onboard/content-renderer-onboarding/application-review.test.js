/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React, {useRef} from "react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import Cookies from "electrode-cookies";
import jquery from "jquery";
import {testStore} from "../../../../../src/client/utility/TestingUtils";
import ApplicationReview from "../../../../../src/client/components/onboard/content-renderer-onboarding/application-review";
import profile from "../../../mocks/userProfile";
import Http from "../../../../../src/client/utility/Http";
import MODALSMETA from "../../../../../src/client/config/modals-meta";
import FORMFIELDMETA from "../../../../../src/client/config/formsConfig/form-field-meta";
import applicationDetails from "../../../mocks/applicationDetails";
import {mount} from "enzyme";

let store;
jest.mock("electrode-cookies");
jest.unmock("jquery")

const setUpRender = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA},
    },
    company: {onboardingDetails: applicationDetails},
    user: {profile}
  };
  store = testStore(mockStore);
  return renderer.create(<Provider store={store}><BrowserRouter><ApplicationReview /></BrowserRouter></Provider>);
};

const setUpMount = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA},
    },
    company: {onboardingDetails: applicationDetails},
    user: {profile}
  };
  store = testStore(mockStore);
  return mount(<Provider store={store}><BrowserRouter><ApplicationReview /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("ApplicationReview test container", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
    Cookies.get.mockImplementation(jest.fn());
  });

  it("should render the ApplicationReview successfully", () => {
    jest.spyOn(Http, "get").mockImplementation(() => Promise.resolve({body: {}}));
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUpRender();
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
