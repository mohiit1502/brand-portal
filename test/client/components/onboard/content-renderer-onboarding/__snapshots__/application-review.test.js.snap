// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationReview test container should render the ApplicationReview successfully 1`] = `
<div
  className="col pl-5 pr-0"
>
  <div
    className="row mt-4 pl-5 ml-5 brand-registration-title font-weight-bold font-size-28"
  >
    <span
      className="col"
    >
      Application Review
    </span>
  </div>
  <div
    className="row mt-3 mb-4 pl-5 ml-5 brand-registration-subtitle"
  >
    <span
      className="col"
    >
      Please ensure the information below is accurate.
    </span>
  </div>
  <div
    className="c-ApplicationDetails text-left row ml-5 pl-5 loader"
  >
    <p
      className="ml-3"
    >
      Getting application details...
    </p>
  </div>
  <div
    className="c-ButtonsPanel form-row py-4 mt-5"
  >
    <div
      className="col company-onboarding-button-panel text-right mr-5"
    >
      <button
        className="btn btn-sm cancel-btn text-primary"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
      <button
        className="btn btn-sm btn-primary submit-btn px-4 ml-3"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Confirm and submit
      </button>
    </div>
  </div>
</div>
`;
